Types.h

```diff
@@ -54,21 +54,25 @@ typedef struct SGameState
 typedef struct SBotConfig
 {
     BotType Type;
-    std::string Name;
-    std::string Path;
+    std::string BotName;
+	std::string RootPath;
+	std::string FileName;
     sc2::Race Race;
 	sc2::Difficulty Difficulty;
+	std::string Args; //Optional arguments
     SBotConfig() {}
-    SBotConfig(BotType InType, std::string InBotName, sc2::Race InBotRace, std::string InBotPath, sc2::Difficulty InDifficulty = sc2::Difficulty::Easy)
+    SBotConfig(const BotType & InType,const std::string & InBotName,const sc2::Race & InBotRace, const std::string & InBotPath, const std::string & InFileName, const sc2::Difficulty & InDifficulty = sc2::Difficulty::Easy, const std::string & InArgs="")
```

Comment:
That's too much now :)
BotType is enum which is int by default and can be safely passed as copy and without the const modifier.
The same with the race and difficulty.